### PR TITLE
BUG: add endfunction, endsubroutine to valid fortran end words

### LINKED ISSUE #14625: BUG: f2py endfunction and endsubroutine parsing
Hi, while testing f2py for some simple examples i stumbled upon an error, which i think i have understood the cause, but i didn't go through the f2py implementation to propose a solution.

I believe the fortran file is not parsed correctly when generating the signature file if functions and subroutines are closed with `endfunction` or `endsubroutine`, without space, which is allowed in the gfortran and compiles correctly.

the code test.f90 builds correctly with either `end subroutine` or `endsubroutine`

but in the latter case, the python module loads the first subroutine (`subb`) but not the second (`subc`)
 
### Reproducing code example:

`test.f90`:
```fortran
module test
implicit none
contains
subroutine subb(k)
  real(8), intent(inout) :: k(:)
  k=k+1
endsubroutine
subroutine subc(w,k)
  real(8), intent(in) :: w(:)
  real(8), intent(out) :: k(size(w))
  k=w+1
endsubroutine
end module
```
`build.py`:
```python
src = open('test.f90','r').read()
from numpy import f2py
f2py.compile(source=src,source_fn='test1854.tmp.f90',extension='.f90',extra_args="--opt='-ffree-line-length-none'",modulename='test1854')
import test1854                                                                                                                                                 
print( test1854.test.subb.__doc__ )
import numpy as np
x=np.array([1,2,3],dtype=np.float)
print( test1854.test.subb(x))
print(x)
print( test1854.test.subc.__doc__ )
print( test1854.test.subc(x))
print(x)
```

### Error message:

`python build.py`:
```
Traceback (most recent call last):
  File "build.py", line 10, in <module>
    print( test1854.test.subc.__doc__ )
AttributeError: 'fortran' object has no attribute 'subc'
```

### Numpy/Python version information:

```
1.16.4 3.6.9 |Anaconda, Inc.| (default, Jul 30 2019, 19:07:31) 
[GCC 7.3.0]
```

### PR DESCRIPTION
**Pull request by mattip**
Test, fix for missing end words in fortran subroutines and functions.
Fixes gh-14625

### CODE REVIEW

File: numpy/f2py/cfuncs.py [hunk 1]
```diff
@@ -542,7 +542,7 @@
     'ARRSIZE'] = '#define ARRSIZE(dims,rank) (_PyArray_multiply_list(dims,rank))'
 cppmacros['OLDPYNUM'] = """\
 #ifdef OLDPYNUM
-#error You need to install NumPy version 13 or higher. See https://scipy.org/install.html
+#error You need to install NumPy version 0.13 or higher. See https://scipy.org/install.html
```
Comment (mattip) [inline numpy/f2py/cfuncs.py]: This was left over from a [comment](https://github.com/numpy/numpy/pull/14231#discussion_r312112493) on gh-14231

File: numpy/f2py/tests/test_crackfortran.py [hunk 2]
```diff
@@ -0,0 +1,39 @@
+from __future__ import division, absolute_import, print_function
+
+import pytest
+
+import numpy as np
+from numpy.testing import assert_array_equal
+from . import util
+
+
+class TestNoSpace(util.F2PyTest):
```
Comment (eric-wieser) [inline numpy/f2py/tests/test_crackfortran.py]: nit: Docstring might be handy here to explain what `NoSpace` means

File: numpy/f2py/tests/test_crackfortran.py [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+          character value
+          character t0
+          t0 = value
+        endfunction
+ 
+    """
+
+    def test_module(self):
+        k = np.array([1, 2, 3], dtype = np.float)
+        w = np.array([1, 2, 3], dtype = np.float)
```
Comment (eric-wieser) [inline numpy/f2py/tests/test_crackfortran.py]: Let's not make #6103 worse ;)
```suggestion
        w = np.array([1, 2, 3], dtype=float)
```

Comment (mattip) [reply numpy/f2py/tests/test_crackfortran.py]: heh, fixing

File: numpy/f2py/tests/test_crackfortran.py (hunk 2 above)
Comment (mattip) [reply numpy/f2py/tests/test_crackfortran.py]: adding

File: numpy/f2py/tests/test_crackfortran.py [hunk 4]
```diff
@@ -25,12 +27,11 @@ class TestNoSpace(util.F2PyTest):
...
           t0 = value
         endfunction
- 
     """
 
     def test_module(self):
-        k = np.array([1, 2, 3], dtype = np.float)
-        w = np.array([1, 2, 3], dtype = np.float)
+        k = np.array([1, 2, 3], dtype = np.float64)
+        w = np.array([1, 2, 3], dtype = np.float64)
```
Comment (eric-wieser) [inline numpy/f2py/tests/test_crackfortran.py]: PEP8 says no spaces around `=` here, as in my suggested change

**Review by eric-wieser — APPROVED** [review]
I'm not fortran expert so can't sign off on the validity of the fortran syntax we're parsing, but the changes to the python look fine.

**Comment by seberg** [comment]
@melissawm maybe you can give a hand at reviewing this?

**Comment by melissawm** [comment]
Sure, just give me a few days since I'm a bit busy at work right now, if that's ok!

**Review by melissawm — APPROVED** [review]
Looks good to me!

**Comment by seberg** [comment]
Thanks everyone for the review!

**Pull request merged**